app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java

```diff
@@ -530,20 +543,54 @@ private void checkLocationPermission() {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
             if (locationManager.isLocationPermissionGranted()) {
                 nearbyNoificationCardView.permissionType = NearbyNoificationCardView.PermissionType.NO_PERMISSION_NEEDED;
-                nearbyNoificationCardView.displayPermissionRequestButton(false);
                 locationManager.registerLocationManager();
             } else {
                 nearbyNoificationCardView.permissionType = NearbyNoificationCardView.PermissionType.ENABLE_LOCATION_PERMISSON;
-                nearbyNoificationCardView.displayPermissionRequestButton(true);
+                // If user didn't selected Don't ask again
+                if (shouldShowRequestPermissionRationale(Manifest.permission.ACCESS_FINE_LOCATION)) {
+                        DialogUtil.showAlertDialog(getActivity(),
+                                getString(R.string.nearby_card_permission_title),
+                                getString(R.string.nearby_card_permission_explanation),
+                                () -> displayYouWontSeeNearbyMessage(),
+                                () -> enableLocationPermission());
+                }
             }
         } else {
             // If device is under Marshmallow, we already checked for GPS
             nearbyNoificationCardView.permissionType = NearbyNoificationCardView.PermissionType.NO_PERMISSION_NEEDED;
-            nearbyNoificationCardView.displayPermissionRequestButton(false);
             locationManager.registerLocationManager();
         }
     }
 
+    private void enableLocationPermission() {
+        if (!getActivity().isFinishing()) {
+            ((MainActivity) getActivity()).locationManager.requestPermissions(getActivity());
+        }
+    }
+
+    private void enableGPS() {
+        new AlertDialog.Builder(getActivity())
+                .setMessage(R.string.gps_disabled)
+                .setCancelable(false)
+                .setPositiveButton(R.string.enable_gps,
+                        (dialog, id) -> {
+                            Intent callGPSSettingIntent = new Intent(
+                                    android.provider.Settings.ACTION_LOCATION_SOURCE_SETTINGS);
+                            Timber.d("Loaded settings page");
+                            ((MainActivity) getActivity()).startActivityForResult(callGPSSettingIntent, 1);
+                        })
+                .setNegativeButton(R.string.menu_cancel_upload, (dialog, id) -> {
+                    dialog.cancel();
+                    displayYouWontSeeNearbyMessage();
+                })
+                .create()
+                .show();
+    }
+
+    private void displayYouWontSeeNearbyMessage() {
+        ViewUtil.showLongToast(getActivity(), "You wont see narby place since you didn't gave permission");
```

Comment:
Please make this a string resource and change it to "Unable to display nearest place that needs pictures without location permissions".